grpc/internal/grpcrand/grpcrand.go

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package grpcrand implements math/rand functions in a concurrent-safe way
+// with a global random source, independent of math/rand's global source.
```

Comment:
This is interesting! Since Go 1.2, math/rand has documented:

> The default Source is safe for concurrent use by multiple goroutines.

and the latest version adds:

> but Sources created by NewSource are not.

Maybe this code predates Go 1.2? At any rate, no action needed here; we should do it the way grpc does it. But I'm kinda curious now to check out the commit history upstream.